### PR TITLE
[513219] Active annotation at method parameter leads to ClassCastException #171

### PR DESCRIPTION
**Pull request by cdietrich**
[513219] Active annotation at method parameter leads to ClassCastException #171

Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=513219

**Comment by cdietrich** [comment]
please review

File: org.eclipse.xtend.core/src/org/eclipse/xtend/core/macro/AnnotationProcessor.xtend [hunk 1]
```diff
@@ -55,7 +55,13 @@ class AnnotationProcessor {
 					registerGlobalsCtx.compilationUnit = ctx.compilationUnit
 					
 					runWithCancelIndiciator(ctx, monitor) [|
-						processor.doRegisterGlobals(ctx.annotatedSourceElements.map[ctx.compilationUnit.toXtendMemberDeclaration(it as XtendMember)], registerGlobalsCtx)
+						processor.doRegisterGlobals(ctx.annotatedSourceElements.map[
+							val xtendMember = switch it {
+								XtendMember : ctx.compilationUnit.toXtendMemberDeclaration(it)
+								XtendParameter : ctx.compilationUnit.toXtendParameterDeclaration(it)
+							}
+							return xtendMember
```
Comment (kthoms) [inline org.eclipse.xtend.core/src/org/eclipse/xtend/core/macro/AnnotationProcessor.xtend]: the `xtendMember` var is not necessary here, or do you want to leave it for debugging purpose?

Comment (cdietrich) [inline org.eclipse.xtend.core/src/org/eclipse/xtend/core/macro/AnnotationProcessor.xtend]: that was the idea

Comment (kthoms) [inline org.eclipse.xtend.core/src/org/eclipse/xtend/core/macro/AnnotationProcessor.xtend]: Ok then

**Pull request merged**